heat/core/signal.py

```diff
@@ -88,34 +119,108 @@ def convolve(a: DNDarray, v: DNDarray, mode: str = "full") -> DNDarray:
     a = a.astype(promoted_type)
     v = v.astype(promoted_type)
 
-    if len(a.shape) != 1 or len(v.shape) != 1:
-        raise ValueError("Only 1-dimensional input DNDarrays are allowed")
-    if mode == "same" and v.shape[0] % 2 == 0:
+    # check if the filter is longer than the signal and swap them if necessary
+    if v.shape[-1] > a.shape[-1]:
+        a, v = v, a
+
+    batch_processing = False
+    if a.ndim > 1:
+        # batch processing requires 1D filter OR matching batch dimensions for signal and filter
+        batch_dims = a.shape[:-1]
+        # verify that the filter shape is consistent with the signal
+        if v.ndim > 1:
+            if v.shape[:-1] != batch_dims:
+                raise ValueError(
+                    f"Batch dimensions of signal and filter must match. Signal: {a.shape}, Filter: {v.shape}"
+                )
+        if a.is_distributed():
+            if a.split == a.ndim - 1:
+                raise ValueError(
+                    "Please distribute the signal along the batch dimension, not the signal dimension. For in-place redistribution use the `DNDarray.resplit_()` method with `axis=0`"
+                )
+            if v.is_distributed():
+                if v.ndim == 1:
+                    # gather filter to all ranks
+                    v.resplit_(axis=None)
+                else:
+                    v.resplit_(axis=a.split)
+        batch_processing = True
```

Comment:
> Why is batch_processing = True hard coded?

Hi @krajsek , do you mean, it would be better to let the user set it as a keyword argument? If so, I agree.

Reply:
As far as I understand things, it is set `batch_processing=True` and, depending on some conditions, it is set `batch_processing=False` lateron; in my opinion this is fine and I would not suggest to introduce a kwarg since actually the correct value of `batch_processing` is already uniquely determined by the other inputs.

Reply:
o.k.,  I agree, I also do not advocate for kwarg. Setting variables to some values in code always irritates me, but after reading the code again it makes sense. Maybe we spend a comment why this is hard coded here.